### PR TITLE
feat(query): predict_linear support

### PR DESCRIPTION
**Pull request by HimaVarsha94**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

### CODE REVIEW

**Comment by velvia** [comment]
@vishramachandran  and @TanviBhavsar have a look

File: query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala [hunk 1]
```diff
@@ -830,3 +830,95 @@ class HoltWintersChunkedFunctionL(funcParams: Seq[FuncArgs]) extends HoltWinters
     }
   }
 }
+
+abstract class PredictLinearChunkedFunction(funcParams: Seq[Any],
```
Comment (velvia) [inline query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala]: Would be great add some docs here.... links or descriptions of what predict linear is, and how the algorithm works

Comment (HimaVarsha94) [reply query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala]: https://github.com/filodb/FiloDB/pull/568/commits/7e03132cf261835bc5f8bc640a0f5860b9f38860

**Comment by velvia** [comment]
Thanks @HimaVarsha94 ... merging

**Pull request merged**